### PR TITLE
Add Localization doc

### PR DESCRIPTION
**Pull request by bc-andreadao**
## What/Why?

- Add a doc for localization

### CODE REVIEW

File: docs/localization.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    "bestSellingProducts": "Produits les plus vendus",
+    "featuredProducts": "Produits populaires",
+    "newestProducts": "Produits les plus recentsProduits"
+  }
+}
+``` 
+
+
+
+### {}
```
Comment (bc-andreadao) [inline docs/localization.md:80]: What are the variables in brackets in the `en.json` file? 

For example, where do {rating} and {author} refer to? 
```
"reviewRating": "Rating: {rating} out of 5 stars",
"reviewAuthor": "Posted by {author} on"
```

File: docs/localization.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+You must add the language translations to the `i18n.ts` file.
+
+For example, if you created a `fr.json` file, include the `fr` [BCP 47 specification](https://tools.ietf.org/html/bcp47) when you define the locales:   
+
+```ts
+const locales = ['en', 'fr'] as const;
+```
+
+## page.tsx file
```
Comment (bc-andreadao) [inline docs/localization.md:94]: I could use some help with this section

File: docs/localization.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+[!Note]
+> #### Default language
+> Catalyst hardcodes the `en.json` file as the default language file, which contains U.S. English translations when you set up the project [using the CLI](https://www.catalyst.dev/docs/cli).
+
+## Translation file
+
+To translate your theme, create a JSON file for each language you choose to support. Each language that you want to support must have its own JSON file. These can include non-U.S. versions of English, each with their own spellings.
+
+Name your translation files based on the [BCP 47 specification](https://tools.ietf.org/html/bcp47) of language tags and region codes. For an overview of how these language tags are constructed, see [Language tags in HTML and XML](http://www.w3.org/International/articles/language-tags/).
```
Comment (bc-andreadao) [inline docs/localization.md:31]: Is this section accurate? I took it off the Stencil docs

File: docs/localization.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
+# Localization 
+
+You can localize your Catalyst storefront so the theme appears in the shopper's preferred language.
+Your storefront displays the language stored in the shopper's browser cookie if you provided translations for that language. 
+Shoppers can then see theme content in the language of their choice throughout their browsing and checkout experience.
```
Comment (bc-andreadao) [inline docs/localization.md:5]: Is "theme content" accurate? It was terminology we used from Stencil

File: docs/localization.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  // ...
+
+  return (
+    <div>
+      <NextIntlClientProvider locale={locale} messages={{ Product: messages.Product ?? {} }}>
+        <ProductCardCarousel
+          products={featuredProducts}
+          title={t('Carousel.featuredProducts')}
+        />
+      </NextIntlClientProvider>
```
Comment (jorgemoya) [inline docs/localization.md:114]: So this example is more aimed at using next-intl inside client components (which in turn use the `useTranslations` hook). I would demo how to us this in a server component (`getTranslations`) and ultimately how to use it in a client component (by using this provider like in this example).

**Review by jorgemoya — COMMENTED** [review]
Hey @BC-krasnoshapka, can someone else take a look at this as well? 🙏

File: docs/localization.md (hunk 5 above)
Comment (bc-andreadao) [reply docs/localization.md:114]: Yes, if someone could demo it to me and/or provide some example snippets that would be great

Comment (BC-krasnoshapka) [reply docs/localization.md:114]: @jorgemoya I've adjusted examples for server and client components. Let me know if it's good now.

Comment (jorgemoya) [reply docs/localization.md:114]: Looks good!

**Review by jorgemoya — APPROVED** [review]
lgtm!

File: docs/localization.md (hunk 3 above)
Comment (bc-alexsaiannyi) [reply docs/localization.md:29]: In my opinion, it's accurate.

File: docs/localization.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+import { LocaleType } from '~/i18n';
+
+interface Props {
+  params: {
+    locale: LocaleType;
+  };
+}
+
+export default async function Home({ params: { locale } }: Props) {
+  unstable_setRequestLocale(locale);
```
Comment (bc-alexsaiannyi) [inline docs/localization.md:101]: @BC-krasnoshapka do we need to pay attention on using `unstable_setRequestLocale` and add a link to `next-intl` section regarding this method?

**Review by bc-alexsaiannyi — APPROVED** [review]
Great job 🥳

**Comment by bc-andreadao** [comment]
Thanks everyone!

**Pull request merged**